pkg/operator/controller/controller.go

```diff
@@ -154,14 +160,19 @@ func (r *reconciler) reconcile(request reconcile.Request) (reconcile.Result, err
 				return result, utilerrors.NewAggregate(errs)
 			}
 
+			ingressDomain, err := effectiveIngressDomain(ingress, ingressConfig)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("failed to find a valid unclaimed ingress domain for clusteringress %s: %v", ingress.Name, err))
```

Comment:
hmm, doesn't this mean in the case of an error we'll proceed to use an empty value (which itself has a meaning)? How can you distinguish between a user-specified empty value and the empty fallback we return with the error?

Should this be fatal to the reconciliation?

Reply:
You're right.  We can continue if `ingressDomain` is empty (which means we won't set the router's canonical hostname, publish DNS records, or generate a default certificate), but we must skip the `ensureIngressDelete`/`ensureClusterIngress` if `err` is not nil.  Will fix...

Reply:
Fixed.  Thanks!